src/map/atcommand.c

```diff
@@ -2798,73 +2802,89 @@ ACMD(hatch)
 	return true;
 }
 
-/*==========================================
+/**
+ * Sets pet's intimacy value.
  *
- *------------------------------------------*/
+ * @code{.herc}
+ *	@petfriendly <0-1000>
+ * @endcode
+ *
+ */
 ACMD(petfriendly)
 {
-	int friendly;
-	struct pet_data *pd;
-
-	if (!*message || (friendly = atoi(message)) < 0) {
-		clif->message(fd, msg_fd(fd,1016)); // Please enter a valid value (usage: @petfriendly <0-1000>).
+	if (*message == '\0' || (atoi(message) == 0 && !isdigit(*message))) {
+		clif->message(fd, msg_fd(fd, 1016)); /// Please enter a valid value (usage: @petfriendly <0-1000>).
 		return false;
 	}
 
-	pd = sd->pd;
-	if (!pd) {
-		clif->message(fd, msg_fd(fd,184)); // Sorry, but you have no pet.
+	int friendly = atoi(message);
+
+	if (friendly < PET_INTIMACY_NONE || friendly > PET_INTIMACY_MAX) {
+		clif->message(fd, msg_fd(fd, 37)); /// An invalid number was specified.
 		return false;
 	}
 
-	if (friendly < 0 || friendly > 1000)
-	{
-		clif->message(fd, msg_fd(fd,37)); // An invalid number was specified.
+	struct pet_data *pd = sd->pd;
+
+	if (sd->status.pet_id == 0 || pd == NULL) {
+		clif->message(fd, msg_fd(fd, 184)); /// Sorry, but you have no pet.
 		return false;
 	}
 
-	if (friendly == pd->pet.intimate) {
-		clif->message(fd, msg_fd(fd,183)); // Pet intimacy is already at maximum.
+	if (friendly == pd->pet.intimate && friendly == PET_INTIMACY_MAX) {
+		clif->message(fd, msg_fd(fd, 183)); /// Pet intimacy is already at maximum.
 		return false;
 	}
 
-	pet->set_intimate(pd, friendly);
-	clif->send_petstatus(sd);
-	clif->message(fd, msg_fd(fd,182)); // Pet intimacy changed.
+	if (friendly != pd->pet.intimate) { /// No need to update the pet's status if intimacy value won't change.
+		pet->set_intimate(pd, friendly);
+		clif->send_petstatus(sd);
+	}
+
+	clif->message(fd, msg_fd(fd, 182)); /// Pet intimacy changed. (Send message regardless of value has changed or not.)
 	return true;
 }
 
-/*==========================================
+/**
+ * Sets pet's hunger value.
  *
- *------------------------------------------*/
+ * @code{.herc}
+ *	@pethungry <0-100>
+ * @endcode
+ *
+ */
 ACMD(pethungry)
 {
-	int hungry;
-	struct pet_data *pd;
-
-	if (!*message || (hungry = atoi(message)) < 0) {
-		clif->message(fd, msg_fd(fd,1017)); // Please enter a valid number (usage: @pethungry <0-100>).
+	if (*message == '\0' || (atoi(message) == 0 && !isdigit(*message))) {
+		clif->message(fd, msg_fd(fd, 1017)); /// Please enter a valid number (usage: @pethungry <0-100>).
 		return false;
 	}
 
-	pd = sd->pd;
-	if (!sd->status.pet_id || !pd) {
-		clif->message(fd, msg_fd(fd,184)); // Sorry, but you have no pet.
+	int hungry = atoi(message);
+
+	if (hungry < PET_HUNGER_NONE || hungry > PET_HUNGER_STUFFED) {
```

Comment:
instead of checking for min/max value, how about just cap the value and let it proceed?
```suggestion
	hungry = cap_value(hungry, PET_HUNGER_NONE, PET_HUNGER_STUFFED);
```

Reply:
Same as above. I think throwing an error message here, to notifiy the user, is the better way,